streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -1136,6 +1144,22 @@ public boolean commitNeeded() {
         }
     }
 
+    @Override
+    public Map<TopicPartition, Long> getCommittedOffsets() {
+        return committedOffsets;
+    }
+
+    @Override
+    public Map<TopicPartition, Long> getHighWaterMark() {
+        highWatermark.putAll(recordCollector.offsets());
+        return highWatermark;
```

Comment:
Seems the `highWatermark` is not used anywhere. Why not returning `recordCollector.offsets()` only?

Reply:
`recordCollector.offsets()` could be missing some entires so we collect them in `highWatermark` if it is called multiple times for the Task which is expected.